app/src/main/java/nu/yona/app/utils/MobileNumberFormatter.java

```diff
@@ -40,4 +42,46 @@ private static String formatDutchNumber(String number) {
         return result;
     }
 
+/*
+        Below function will do following on mobile number:
+            - Remove white spaces, new lines any other unwanted characters our of character set "[^0123456789+]"
+            - Concate country code prefix if it is not present.
+            - Add the prefix from the prefix input field if the number doesn't already start with +
+*/
+
+    public static String formatMobileNumber(String countryCode, String mobileNumber){
+        countryCode = removeUnwantedCharacters(countryCode);
+        mobileNumber = removeUnwantedCharacters(mobileNumber);
+        String mobileNumberWithCountryCode = mobileNumber;
+        if(!mobileNumber.startsWith("+")){
+            if(!countryCode.startsWith("+")){
+                countryCode = "+"+countryCode;
+            }
+            mobileNumberWithCountryCode = countryCode+mobileNumber;
+        }
+        return formatMobileNumberWithCountryCode(mobileNumberWithCountryCode);
+    }
+
+    /*
+        Below function will do following on mobile number:
+             - Remove white spaces, new lines any other unwanted characters our of character set "[^0123456789+]"
+            - Remove the construct (0) if that occurs in the number
+            - Replace +310 with +31, as it's a common Dutch mistake to convert a mobile number like 06 1819... to +31061819...
+            - Validate the regex "^\\+[0-9]{6,20}$"
+*/
+
+    public static String formatMobileNumberWithCountryCode(String mobileNumberWithCountryCode){
+        mobileNumberWithCountryCode = removeUnwantedCharacters(mobileNumberWithCountryCode);
+        mobileNumberWithCountryCode = mobileNumberWithCountryCode.replace("(0)","");
+        if(mobileNumberWithCountryCode.startsWith("+310")){
+            mobileNumberWithCountryCode = mobileNumberWithCountryCode.replace("+310","+31");
+        }
+        return mobileNumberWithCountryCode;
+    }
+
+    // To validate after formatting the number.
+    public static boolean validateMobileNumber(String mobileNumber){
```

Comment:
In line with the other rename requests, rename this one to simply ``validate``. It's on the ``MobileNumberFormatter``, so it's clear it validates a mobile number.